ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -724,16 +722,17 @@ - (unsigned long) getUsedMemory {
         __strong RNDeviceInfo *strongSelf = weakSelf;
         if (strongSelf) {
             // Save WKWebView (it might deallocate before we ask for user Agent)
-            strongSelf->webView = [[WKWebView alloc] init];
+            __block WKWebView *webView = [[WKWebView alloc] init];
 
-            [strongSelf->webView evaluateJavaScript:@"window.navigator.userAgent;" completionHandler:^(id _Nullable result, NSError * _Nullable error) {
+            [webView evaluateJavaScript:@"window.navigator.userAgent;" completionHandler:^(id _Nullable result, NSError * _Nullable error) {
                 if (error) {
                     reject(@"getUserAgentError", error.localizedDescription, error);
+                    webView = nil
                     return;
                 }
                 resolve([NSString stringWithFormat:@"%@", result]);
                 // Destroy the WKWebView after task is complete
-                strongSelf->webView = nil;
+                webView = nil;
```

Comment:
Interesting! My only comment is that now we are "cleaning up" (`webView = nil`) in two places, and I strongly prefer to clean up in one place only if possible.

I think the `webView = nil` could go on top of the `if (error) {` as we no longer need the webView anymore do we? Then there is just one `webView = nil`

Reply:
😀
It's a good idea. 

----
In fact, before submitting pr, I also want to do this, but I want to try not to change the code
So.....